pkg/apis/planetscale/v2/vitesscell_types.go

```diff
@@ -117,12 +119,42 @@ type VitessCellImages struct {
 	Vtgate string `json:"vtgate,omitempty"`
 }
 
+// AutoscalerSpec defines the vtgate's pod autoscaling specification.
+type AutoscalerSpec struct {
+	// MinReplicas is the minimum number of instances of vtgate to run in
+	// this cell when autoscaling is enabled.
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	MinReplicas *int32 `json:"minReplicas,omitempty"`
+
+	// MaxReplicas is the maximum number of instances of vtgate to run in
+	// this cell when autoscaling is enabled.
+	// +kubebuilder:validation:Minimum=0
+	MaxReplicas int32 `json:"maxReplicas,omitempty"`
+
+	// Metrics is meant to provide a customizable way to configure HPA metrics.
+	// currently the only supported custom metrics is type=Pod.
+	// Use TargetCPUUtilization or TargetMemoryUtilization instead if scaling on these common resource metrics.
+	// +listType=atomic
+	// +optional
+	Metrics []autoscalingv2.MetricSpec `json:"metrics,omitempty"`
+
+	// Behavior specifies the scaling behavior of the target in both Up and Down directions.
+	// +optional
+	Behavior *autoscalingv2.HorizontalPodAutoscalerBehavior `json:"behavior,omitempty"`
+}
```

Comment:
@frouioui Let me know if I need to make any changes to this spec. It is similar to [HorizontalPodAutoscaler](https://pkg.go.dev/k8s.io/api/autoscaling/v2#HorizontalPodAutoscalerSpec), without the ScaleTargetRef field, whose value I am setting in https://github.com/siadat/vitess-operator/blob/sina/vitesscell-hpa/pkg/operator/vtgate/horizontal_pod_autoscaler.go#L46-L50

Reply:
I think this is fine. If you don't want the end-user to customize `ScaleTargetRef`, which makes sense as it should be constant, then having our own `AutoscalerSpec` is okay.